client/src/components/UpdateFieldAgent.js

```diff
@@ -0,0 +1,109 @@
+import { useState } from "react";
+
+function UpdateFieldAgent({ fieldAgent, handleUpdate, handleCancel }) {
```

Comment:
Challenge for you: do you think there's a way to have a single component that both adds and updates?